pkg/monitoring/docker/docker.go

```diff
@@ -22,10 +23,17 @@ func ListContainers() (map[string]interface{}, error) {
 			return nil, err
 		}
 	}
+	var dockerExecExists bool
+	if _, err := exec.LookPath("docker"); err == nil {
+		dockerExecExists = true
+	}
 
 	containers, err := client.ListContainers(docker.ListContainersOptions{All: false})
-	if err != nil && err == docker.ErrConnectionRefused && !connectionSucceedOnce {
-		// do not produce error if Docker is missing and wasn't successfully connected in this session before
+	if err != nil &&
```

Comment:
Can we simplify this statement? Maybe split into two `if`?
Is the check against `connectionSucceedOnce` necessary? It's only declared but never accessed until this point

Reply:
connectionSucceedOnce is set to true in case of successful docker query

Reply:
> connectionSucceedOnce is set to true in case of successful docker query

But only at line 44 which is _after_ this check.

Reply:
Yes, I know how it works and tried to do this way at first but
- I needed to store this struct at the Cagent struct
- I needed to rework `processesFromProc()` and `processes()` to pass the Cagent struct in order to run `ContainerNameByID`. I think it is not clear to use methods when it is possible to solve the task with just a func

Reply:
> With the post-inspection finding nothing to cause the engines to fail, engineers are saying the likely cause for the shutdown is a software issue onboard the aircraft.

What exactly is not clear?

Reply:
> And in this case, the price is too high

Which price exactly?

Reply:
I've just listed the reasons :)
`processes()` and `processesFromProc()` are clear and independent as they don't need to have Cagent instance passed in order to provide results.

DO you think it is ok to rewrite them all to become methods?

Reply:
> DO you think it is ok to rewrite them all to become methods?

yes. That's why I did the `Top` package how I did it. No global state, a central struct, `Top` in this case, which can hold state and other private members but still nicely groups the whole functionality together. If you're looking for anything related it's all within that one package and not together with 10+ other files in one package without a struct that holds things together.

In the end the same applies for `Config`, `CPUWatcher` etc.

Reply:
take a look. I've just reworked it